pkg/system/phase2_creating.go

```diff
@@ -22,6 +24,17 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var oauthClient *http.Client
```

Comment:
Let's save the client on the reconciler, global var seems wrong.

Reply:
@nb-ohad @romayalon Wouldn't that cause the same problem though? The reconciler object gets created on every reconcile...

Reply:
Yes, you are right.